packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -51,26 +54,24 @@ function BlockListAppender( {
 		);
 	}
 
-	// IgnoreNestedEvents is used to treat interactions within the appender as
-	// subject to the same conditions as those which occur within nested blocks.
-	// Notably, this effectively prevents event bubbling to block ancestors
-	// which can otherwise interfere with the intended behavior of the appender
-	// (e.g. focus handler on the ancestor block).
-	//
-	// A `tabIndex` is used on the wrapping `div` element in order to force a
-	// focus event to occur when an appender `button` element is clicked. In
-	// some browsers (Firefox, Safari), button clicks do not emit a focus event,
-	// which could cause this event to propagate unexpectedly. The `tabIndex`
-	// ensures that the interaction is captured as a focus, without also adding
-	// an extra tab stop.
-	//
-	// See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
 	return (
-		<IgnoreNestedEvents childHandledEvents={ [ 'onFocus', 'onClick', 'onKeyDown' ] }>
-			<div tabIndex={ -1 } className="block-list-appender">
-				{ appender }
-			</div>
-		</IgnoreNestedEvents>
+		<div
+			// A `tabIndex` is used on the wrapping `div` element in order to
+			// force a focus event to occur when an appender `button` element
+			// is clicked. In some browsers (Firefox, Safari), button clicks do
+			// not emit a focus event, which could cause this event to propagate
+			// unexpectedly. The `tabIndex` ensures that the interaction is
+			// captured as a focus, without also adding an extra tab stop.
+			//
+			// See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
+			tabIndex={ -1 }
+			// Prevent the block from being selected when the appender is
+			// clicked.
+			onFocus={ stopPropagation }
```

Comment:
@aduth This fixed the issue you found (https://github.com/WordPress/gutenberg/pull/19397#issuecomment-570602468). Does everything now work correctly?

Reply:
Initial impression: I imagine this could work, yes, considering that `IgnoreNestedEvents` was in many ways a glorified `stopPropagation` anyways. Will test as part of a broader re-review.

Reply:
Thanks! It would be nice the eventually get rid of this as well. I'll look into it sometime separately.

Reply:
> Thanks! It would be nice the eventually get rid of this as well. I'll look into it sometime separately.

Yeah, that's fair. I'm very reluctant about any `stopPropagation`, but I think it's fair to say it's not really "introduced" here, since it was more-or-less the same effect previously.

Reply:
Right, same here. At least it's now explicit that propagation is stopped here. `IgnoreNestedEvents` kind of made it look like it's more ok.

Reply:
> Right, same here. At least it's now explicit that propagation is stopped here. `IgnoreNestedEvents` kind of made it look like it's more ok.

It's a give and take, I think. One nice thing about `IgnoreNestedEvents` is that it wouldn't stop _all_ propagation, only that which it expressly cares about, thus potentially avoiding issues with other (intended) event handling further up the element hierarchy.